### ISSUE TITLE
src folder reorganization

### ISSUE THREAD

**Issue by MrSeljak** [CleverRaven/Cataclysm-DDA#26556]
I've been having fun with this game for quite some time and after I found out it is open source I considered contributing to it. Immediately after opening the src directory it seemed quite chaotic to me. This isn't really a big gripe, but it makes navigating the source files a bit hard.

This could be alleviated a bit by moving files into separate folders within the src folder such as: map_gen folder, NPC folder, map_mechanics, player_mechanics, etc.

It really isn't a big change but it could help newcomers like me to find their way around the code.

**Comment by ZhilkinSerg** [comment]
I agree the code structure may not be optimal, but I do not see if adding folders will help it much. Anyway would love to see some suggestions.

**Comment by Theundyingcode** [comment]
One of the many IDE plugins I have lets me create "virtual" folders so when i'm editing things everything is nice and organized but inreality src remains a huge long list of files.

**Comment by Theundyingcode** [comment]
ummm..... i'd have to go look I got it a long time ago...
